### PR TITLE
Enable stream-style-serialization by default

### PR DESCRIPTION
**Pull request by alzimmermsft**
Enables `stream-style-serialization` by default for Swagger and TypeSpec code generation. Also removes `enable-xml` as it isn't used anymore.

### CODE REVIEW

**Comment by weidongxu-microsoft** [comment]
One reminding issue before we can do the switch is the missing `@Generated` annotation on "toJson" "fromJson" method
https://github.com/Azure/autorest.java/blob/main/typespec-tests/src/main/java/com/_specs_/azure/core/basic/models/User.java#L132-L197

@haolingdong-msft

**Comment by weidongxu-microsoft** [comment]
It could depends on the decision in https://github.com/Azure/autorest.java/issues/2468 as well.
(json-merge-patch in stream-style model would generate code with static block, and we need to decide on how partial-update handles that)

**Comment by haolingdong-msft** [comment]
> One reminding issue before we can do the switch is the missing `@Generated` annotation on "toJson" "fromJson" method https://github.com/Azure/autorest.java/blob/main/typespec-tests/src/main/java/com/_specs_/azure/core/basic/models/User.java#L132-L197
> 
> @haolingdong-msft
> 
> PS: we probably need to let openai still use Jackason due to the "missing 'kind'" discussion (or we may need to customize the "fromJson" for them to handle the case).

Yes, Hi @alzimmermsft, will you fix this, or you want me to fix this? (I have not fixed it currently as you said you will add this) https://github.com/Azure/autorest.java/pull/2410#discussion_r1408712850 Thanks!

**Comment by alzimmermsft** [comment]
Thanks @haolingdong-msft!

It appears this PR needs some more work as the types used in Management plane generation don't extend JsonSerializable (azure-core-management types). So, I'll mark those as `--stream-style-serialization false` to prevent those libraries from failing build. Additional to that, `azure-xml` is still in beta so I may add a switch specific for that, or we can just say it's an all or nothing for now.

**Comment by weidongxu-microsoft** [comment]
I've set `stream-style-serialization` to always `false` on mgmt flow.
- https://github.com/Azure/autorest.java/pull/2506/commits/921cbb206cde4abaefbcf588affea5bfe9a5a966
- https://github.com/Azure/autorest.java/pull/2506/commits/e4b9f381049d3c416686a76f7ee0fe849ad4f1dc

It is not tested there, so we don't want to switch it on there, at present.

**Review by srnagar — APPROVED** [review]

File: typespec-tests/src/main/java/com/_specs_/azure/clientgenerator/core/access/implementation/models/AbstractModel.java [hunk 1]
```diff
@@ -47,6 +75,7 @@ public String getName() {
     public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/main/java/com/_specs_/azure/clientgenerator/core/access/implementation/models/AbstractModel.java:75]: @alzimmermsft @haolingdong-msft @srnagar 

We need `@Generated` on this method.

Else, partial-update would be in a mess.

File: typespec-tests/src/main/java/com/_specs_/azure/clientgenerator/core/access/implementation/models/AbstractModel.java [hunk 2]
```diff
@@ -56,8 +85,7 @@ public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
      * @param jsonReader The JsonReader being read.
      * @return An instance of AbstractModel if the JsonReader was pointing to an instance of it, or null if it was
      * pointing to JSON null.
-     * @throws IllegalStateException If the deserialized JSON object was missing any required properties or the
-     * polymorphic discriminator.
+     * @throws IllegalStateException If the deserialized JSON object was missing any required properties.
      * @throws IOException If an error occurs while reading the AbstractModel.
      */
     public static AbstractModel fromJson(JsonReader jsonReader) throws IOException {
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/main/java/com/_specs_/azure/clientgenerator/core/access/implementation/models/AbstractModel.java:91]: and this

File: typespec-tests/src/main/java/com/_specs_/azure/clientgenerator/core/access/implementation/models/AbstractModel.java [hunk 3]
```diff
@@ -88,17 +116,23 @@ public static AbstractModel fromJson(JsonReader jsonReader) throws IOException {
     static AbstractModel fromJsonKnownDiscriminator(JsonReader jsonReader) throws IOException {
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/main/java/com/_specs_/azure/clientgenerator/core/access/implementation/models/AbstractModel.java:116]: and this

File: typespec-tests/src/main/java/com/_specs_/azure/clientgenerator/core/access/implementation/models/AbstractModel.java [hunk 4]
```diff
@@ -30,9 +36,31 @@ public class AbstractModel implements JsonSerializable<AbstractModel> {
...
+        return this.kind;
+    }
+
+    /**
+     * Set the kind property: The kind property.
+     * 
+     * @param kind the kind value to set.
+     * @return the AbstractModel object itself.
+     */
+    AbstractModel setKind(String kind) {
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/main/java/com/_specs_/azure/clientgenerator/core/access/implementation/models/AbstractModel.java:59]: and this

**Pull request merged**